src/Catalog/SortingIdVersionCollector.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Net.Http;
+using Newtonsoft.Json.Linq;
+using NuGet.Services.Metadata.Catalog.Helpers;
+
+namespace NuGet.Services.Metadata.Catalog
+{
+    public abstract class SortingIdVersionCollector : SortingCollector<FeedHelpers.FeedPackageIdentity>
+    {
+        public SortingIdVersionCollector(Uri index, Func<HttpMessageHandler> handlerFunc = null) : base(index, handlerFunc)
+        {
+        }
+
+        protected override FeedHelpers.FeedPackageIdentity GetKey(JObject item)
+        {
+            return new FeedHelpers.FeedPackageIdentity(item["nuget:id"].ToString(), item["nuget:version"].ToString());
```

Comment:
This won't work well with comparison, since it will be compared by ref, and not by value